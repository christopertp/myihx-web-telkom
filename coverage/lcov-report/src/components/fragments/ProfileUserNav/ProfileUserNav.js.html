
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/fragments/ProfileUserNav/ProfileUserNav.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> / <a href="index.html">src/components/fragments/ProfileUserNav</a> ProfileUserNav.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">80.95% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>17/21</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">62.5% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>5/8</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">81.82% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/11</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">83.33% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>15/18</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a>
<a name='L105'></a><a href='#L105'>105</a>
<a name='L106'></a><a href='#L106'>106</a>
<a name='L107'></a><a href='#L107'>107</a>
<a name='L108'></a><a href='#L108'>108</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { createContext, useContext, useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import Button from '../../elements/Button';
import Modal from '../../elements/Modal';
import { thousand } from '../../../utils/format';
import ActivityCards from '../../elements/ActivityCards';
import styles from './styles.scoped.css';
&nbsp;
const Context = createContext({});
&nbsp;
export function <span class="fstat-no" title="function not covered" >Rewards(</span>) {
  const ACTIVE = <span class="cstat-no" title="statement not covered" >'ACTIVE';</span>
  const { status, poin_total } = <span class="cstat-no" title="statement not covered" >useSelector(<span class="fstat-no" title="function not covered" >s </span>=&gt; <span class="cstat-no" title="statement not covered" >s.profile)</span>;</span>
<span class="cstat-no" title="statement not covered" >  return (</span>
    ACTIVE === status
      ? (
        &lt;Link className={styles.points} to="/profile"&gt;
          &lt;small&gt;IndiHome Points&lt;/small&gt;
          &lt;strong&gt;{thousand(poin_total)}&lt;/strong&gt;
          &lt;img alt="chevron red" src="/assets/ic_chevron_red.svg" /&gt;
        &lt;/Link&gt;
      )
      : (
        &lt;Link className={styles.activate} to="activate-rewards"&gt;
          &lt;img alt="wallet" src="/assets/ic_selection_rewards.svg" /&gt;
          &lt;strong&gt;Activate Rewards&lt;/strong&gt;
          &lt;img alt="chevron red" src="/assets/ic_chevron_red.svg" /&gt;
        &lt;/Link&gt;
      )
  );
}
&nbsp;
export default function ProfileUserNav() {
  const { isLoading, fullName, profilePicture, card } = useSelector(s =&gt; s.profile);
&nbsp;
  return (
    &lt;&gt;
      &lt;section className={styles.identity} &gt;
        &lt;figure style={{ backgroundImage: `url(${profilePicture}` }} /&gt;
        &lt;p&gt;{fullName}&lt;/p&gt;
      &lt;/section&gt;
      &lt;section className={styles.balance}&gt;
        {isLoading.t ? &lt;span className="loading" /&gt; : &lt;TmoneyBalance /&gt;}
        &lt;Rewards /&gt;
      &lt;/section&gt;
      {card &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >&lt;ActivityCards cards={card} fullWidth heading="Upcoming" isLoading={isLoading.c} /&gt;}</span>
    &lt;/&gt;
  );
}
&nbsp;
export function TmoneyBalance() {
  const [blocked, setBlocked] = useState(false);
&nbsp;
  const { balance, status } = useSelector(s =&gt; s.profile).tmoney;
  const configs = {
    BLOCKED: { className: styles.blocked, label: 'Account Blocked', to: '#', onClick: () =&gt; setBlocked(true) },
    ERROR: { className: styles.blocked, label: 'Temporarily Unavailable', to: '#' },
    REGISTERED: { className: styles.active, label: '', to: '/tmoney' },
    UNLINK: { className: '', label: 'Activate Wallet', to: '/tmoney/link' },
    UNREGISTERED: { className: '', label: 'Activate Wallet', to: '/tmoney/activate' },
  };
&nbsp;
  const contextValues = {
    blocked, setBlocked,
  };
&nbsp;
  const { className, label, to, onClick } = configs[status];
&nbsp;
  return (
    &lt;Context.Provider value={contextValues}&gt;
      &lt;Link className={className} onClick={onClick} to={to}&gt;
        &lt;img alt="wallet" src="/assets/ic_selection_wallet.svg" /&gt;
        {
          status === 'REGISTERED' ?
            &lt;&gt;
              &lt;small&gt;Saldo&lt;/small&gt;
              &lt;strong&gt;Rp{thousand(balance)}&lt;/strong&gt;
            &lt;/&gt; :
            &lt;span&gt;{label}&lt;/span&gt;
        }
        &lt;img alt="chevron red" src="/assets/ic_chevron_red.svg" /&gt;
      &lt;/Link&gt;
      &lt;TmoneyBlocked /&gt;
    &lt;/Context.Provider&gt;
  );
}
&nbsp;
export function TmoneyBlocked() {
  const { blocked, setBlocked } = useContext(Context);
  return (
    &lt;Modal className={styles.dialog} onClose={() =&gt; setBlocked(false)} open={blocked}&gt;
      &lt;img src="/assets/grfx_warning.svg" /&gt;
      &lt;h1&gt;Account Blocked&lt;/h1&gt;
      &lt;p&gt;
        Your account has been blocked for your safety.&lt;br /&gt;
        Please contact us to unblock your account.&lt;br /&gt;
        Call &lt;strong&gt;+62 21 3808888&lt;/strong&gt; or&lt;br /&gt;
        email customer &lt;strong&gt;service@tmoney.co.id&lt;/strong&gt;.
      &lt;/p&gt;
      &lt;div&gt;
        &lt;Button onClick={() =&gt; setBlocked(false)} variant="bordered"&gt;Close&lt;/Button&gt;
        &lt;Button onClick={() =&gt; { window.location.href = 'mailto:service@tmoney.co.id'; }}&gt;Mail&lt;/Button&gt;
      &lt;/div&gt;
    &lt;/Modal&gt;
  );
}
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Aug 10 2020 20:49:57 GMT+0700 (Western Indonesia Time)
            </div>
        </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    