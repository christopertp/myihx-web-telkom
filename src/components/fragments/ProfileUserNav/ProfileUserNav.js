import React, { createContext, useContext, useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import Button from '../../elements/Button';
import Modal from '../../elements/Modal';
import { thousand } from '../../../utils/format';
import ActivityCards from '../../elements/ActivityCards';
import styles from './styles.scoped.css';

const Context = createContext({});

export function Rewards() {
  const ACTIVE = 'ACTIVE';
  const { status, poin_total } = useSelector(s => s.profile);
  return (
    ACTIVE === status
      ? (
        <Link className={styles.points} to="/profile">
          <small>IndiHome Points</small>
          <strong>{thousand(poin_total)}</strong>
          <img alt="chevron red" src="/assets/ic_chevron_red.svg" />
        </Link>
      )
      : (
        <Link className={styles.activate} to="activate-rewards">
          <img alt="wallet" src="/assets/ic_selection_rewards.svg" />
          <strong>Activate Rewards</strong>
          <img alt="chevron red" src="/assets/ic_chevron_red.svg" />
        </Link>
      )
  );
}

export default function ProfileUserNav() {
  const { isLoading, fullName, profilePicture, card } = useSelector(s => s.profile);

  return (
    <>
      <section className={styles.identity} >
        <figure style={{ backgroundImage: `url(${profilePicture}` }} />
        <p>{fullName}</p>
      </section>
      <section className={styles.balance}>
        {isLoading.t ? <span className="loading" /> : <TmoneyBalance />}
        <Rewards />
      </section>
      {card && <ActivityCards cards={card} fullWidth heading="Upcoming" isLoading={isLoading.c} />}
    </>
  );
}

export function TmoneyBalance() {
  const [blocked, setBlocked] = useState(false);

  const { balance, status } = useSelector(s => s.profile).tmoney;
  const configs = {
    BLOCKED: { className: styles.blocked, label: 'Account Blocked', to: '#', onClick: () => setBlocked(true) },
    ERROR: { className: styles.blocked, label: 'Temporarily Unavailable', to: '#' },
    REGISTERED: { className: styles.active, label: '', to: '/tmoney' },
    UNLINK: { className: '', label: 'Activate Wallet', to: '/tmoney/link' },
    UNREGISTERED: { className: '', label: 'Activate Wallet', to: '/tmoney/activate' },
  };

  const contextValues = {
    blocked, setBlocked,
  };

  const { className, label, to, onClick } = configs[status];

  return (
    <Context.Provider value={contextValues}>
      <Link className={className} onClick={onClick} to={to}>
        <img alt="wallet" src="/assets/ic_selection_wallet.svg" />
        {
          status === 'REGISTERED' ?
            <>
              <small>Saldo</small>
              <strong>Rp{thousand(balance)}</strong>
            </> :
            <span>{label}</span>
        }
        <img alt="chevron red" src="/assets/ic_chevron_red.svg" />
      </Link>
      <TmoneyBlocked />
    </Context.Provider>
  );
}

export function TmoneyBlocked() {
  const { blocked, setBlocked } = useContext(Context);
  return (
    <Modal className={styles.dialog} onClose={() => setBlocked(false)} open={blocked}>
      <img src="/assets/grfx_warning.svg" />
      <h1>Account Blocked</h1>
      <p>
        Your account has been blocked for your safety.<br />
        Please contact us to unblock your account.<br />
        Call <strong>+62 21 3808888</strong> or<br />
        email customer <strong>service@tmoney.co.id</strong>.
      </p>
      <div>
        <Button onClick={() => setBlocked(false)} variant="bordered">Close</Button>
        <Button onClick={() => { window.location.href = 'mailto:service@tmoney.co.id'; }}>Mail</Button>
      </div>
    </Modal>
  );
}
