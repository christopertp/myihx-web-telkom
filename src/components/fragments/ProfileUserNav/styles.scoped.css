.identity {
  display: flex;
  align-items: center;
  margin-bottom: 1.5rem;
}

.identity > figure {
  background-color: #f2f2f2;
  background-repeat: no-repeat;
  background-size: cover;
  border: 0.125rem solid #e5e5e5;
  border-radius: 50%;
  height: 4rem;
  margin-right: 1rem;
  width: 4rem;
}

.identity > p {
  color: #000001;
  font-size: 1.25rem;
  font-weight: 800;
}

.balance {
  display: flex;
  justify-content: space-between;
  margin-bottom: 1.5rem;
}

.balance > a {
  background-color: #ffffff;
  border-radius: 0.5rem;
  box-shadow: 0 0.25rem 0.5rem rgba(0, 0, 1, 0.075);
  display: grid;
  flex: 1;
  grid-template-rows: auto auto;
  max-width: 50%;
  padding: 1.25rem 1rem;
}

.balance > a > small {
  align-self: self-end;
  color: #999999;
  font-size: 90%;
}

.balance > a > strong {
  color: #333435;
  font-size: 1.125rem;
  grid-column: 1;
}

.balance > a > img, .activate > strong {
  align-self: center;
  grid-row: 1/3;
}

.balance > a > img:last-child {
  grid-column: 2;
  height: 1.2rem;
}

.balance > a:first-child, .activate {
  margin-right: 1rem;
  grid-template-columns: auto 1fr auto;
}

.activate {
  margin-right: 0;
}

.balance > a:first-child > img:first-child, .activate > img:first-child {
  height: 3.125rem;
  margin-right: 1rem;
  width: 3.125rem;
}

.balance > a:first-child > strong, .activate > strong { grid-column: 2!important; }

.balance > a:first-child > img:last-child, .activate > img:last-child { grid-column: 3!important; }

.points {
  grid-template-columns: 1fr auto;
}

.balance > a > span {
  align-self: center;
  color: #333333;
  font-size: 0.9375rem;
  font-weight: bold;
  grid-row: 1/3;
  line-height: 1.1875rem;
}

.balance > a.blocked > img:last-child {
  display: none;
}

.balance > span {
  border-radius: 0.5rem;
  height: 6rem;
  width: 48%;
}

.dialog {
  text-align: center;
}

.dialog > button {
  margin-top: 1.3125rem;
  width: 100%;
}

.dialog > div {
  display: grid;
  grid-template-columns: 1fr 1fr;
  margin-top: 2.5625rem;
}

.dialog > div > button:first-child {
  margin-right: 0.4375rem;
}

.dialog > div > button:last-child {
  margin-left: 0.4375rem;
}

.dialog > h1 {
  color: #191a1a;
  font-size: 1.25rem;
  font-weight: bold;
  line-height: 1.625rem;
  margin-bottom: 0.5rem;
}

.dialog > p {
  color: #191a1a;
  font-size: 0.8125rem;
  line-height: 1.1875rem;
}

@media only screen and (max-width: 48em) {
  .identity { margin-top: 0.5rem; }
  .identity > img {
    border-radius: 50%;
    height: 2.625rem;
    margin-right: 0.75rem;
    width: 2.625rem;
  }
  .identity > p { font-size: 1rem; }
  .balance > a { padding: 0.375rem; }
  .balance > a.active { padding: 0.8125rem 1rem; }
  .balance > a:first-child {
    grid-template-columns: 1fr auto;
  }
  .balance > a:first-child > strong { grid-column: 1; }
  .balance > a.active > img:first-child { display: none;}
  .balance > a:first-child > img:first-child { margin-right: 0; }
  .balance > a > small { font-size: 0.75rem; }
  .balance > a > strong { font-size: 1rem; }
  .balance > span { height: 3rem; }
  .completion { padding: 0.75rem; }
  .completion > h5 { font-size: 1rem; }
  .completion > span { font-size: 0.8rem; }
}
