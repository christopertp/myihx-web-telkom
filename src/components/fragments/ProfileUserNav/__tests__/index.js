import React, { useContext, useState } from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import ProfileUserNav, { TmoneyBalance, TmoneyBlocked } from '../ProfileUserNav';

useSelector.mockImplementation(fn => {
  fn({});
  return {
    isLoading: { t: false },
    fullname: '',
    profilePicture: '',
    tmoney: { extraData: { balance: 0 }, message: '', status: 'BLOCKED' },
  };
});

describe('src/components/fragments/ProfileUserNav', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ProfileUserNav />);
    expect(tree).toMatchSnapshot();
  });

  test('TmoneyBalance', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<TmoneyBalance />);
    expect(tree).toMatchSnapshot();
  });

  test('TmoneyBlocked', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<TmoneyBlocked />);
    expect(tree).toMatchSnapshot();
  });

  test('TmoneyBalance loading', () => {
    useSelector.mockImplementationOnce(() => ({ isLoading: { t: true } }));
    const result = ProfileUserNav();
    expect(result.props.children[1].props.children[0].props.className).toBe('loading');
  });

  test('TmoneyBalance status blocked', () => {
    const setBlocked = jest.fn();
    const tmoney = { balance: 0, message: '', status: 'BLOCKED' };
    useSelector.mockImplementationOnce(() => ({ tmoney }));
    useState.mockImplementationOnce(() => [false, setBlocked]);

    const result = TmoneyBalance();
    result.props.children[0].props.onClick();
    expect(setBlocked).toHaveBeenCalledWith(true);
  });

  test('TmoneyBalance status registered', () => {
    const tmoney = { balance: 10000000, message: '', status: 'REGISTERED' };
    useSelector.mockImplementationOnce(() => ({ tmoney }));

    const result = TmoneyBalance();
    expect(result.props.children[0].props.children[1].props.children[1].props.children.join('')).toBe('Rp10.000.000');
  });

  test('TmoneyBlocked set blocked', () => {
    const setBlocked = jest.fn();
    useContext.mockImplementationOnce(() => ({ blocked: true, setBlocked }));

    const result = TmoneyBlocked();
    result.props.onClose();
    expect(setBlocked).toHaveBeenCalledTimes(1);

    result.props.children[3].props.children[0].props.onClick();
    expect(setBlocked).toHaveBeenCalledTimes(2);

    global.window = Object.create(window);
    Object.defineProperty(window, 'location', { value: { href: '' } });
    result.props.children[3].props.children[1].props.onClick();
    expect(window.location.href).toEqual('mailto:service@tmoney.co.id');
  });
});
