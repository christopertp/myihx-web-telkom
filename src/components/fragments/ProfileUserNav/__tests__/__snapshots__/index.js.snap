// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fragments/ProfileUserNav TmoneyBalance 1`] = `
<Provider
  value={
    Object {
      "blocked": false,
      "setBlocked": [MockFunction],
    }
  }
>
  <Link
    onClick={[Function]}
    to="#"
  >
    <img
      alt="wallet"
      src="/assets/ic_selection_wallet.svg"
    />
    <span>
      Account Blocked
    </span>
    <img
      alt="chevron red"
      src="/assets/ic_chevron_red.svg"
    />
  </Link>
  <TmoneyBlocked />
</Provider>
`;

exports[`src/components/fragments/ProfileUserNav TmoneyBlocked 1`] = `
<Modal
  className=""
  onClose={[Function]}
  open={false}
>
  <img
    src="/assets/grfx_warning.svg"
  />
  <h1>
    Account Blocked
  </h1>
  <p>
    Your account has been blocked for your safety.
    <br />
    Please contact us to unblock your account.
    <br />
    Call 
    <strong>
      +62 21 3808888
    </strong>
     or
    <br />
    email customer 
    <strong>
      service@tmoney.co.id
    </strong>
    .
  </p>
  <div>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      to=""
      type="button"
      variant="bordered"
    >
      Close
    </Button>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      to=""
      type="button"
      variant="red"
    >
      Mail
    </Button>
  </div>
</Modal>
`;

exports[`src/components/fragments/ProfileUserNav render 1`] = `
<React.Fragment>
  <section>
    <figure
      style={
        Object {
          "backgroundImage": "url(",
        }
      }
    />
    <p />
  </section>
  <section>
    <TmoneyBalance />
    <Rewards />
  </section>
</React.Fragment>
`;
